packages/ragbits-document-search/src/ragbits/document_search/documents/sources.py

```diff
@@ -86,6 +89,7 @@ def get_id(self) -> str:
         """
         return f"gcs:gs://{self.bucket}/{self.object_name}"
 
+    @requires_dependencies(["gcloud.aio.storage"], "gcs")
```

Comment:
I am testing a new way of handling optional dependencies. Since managing error messages from extra deps is starting to get tedious, I decided to wrap them in `@requires_dependencies` decorator that handles them automatically. 

If you like it, we can add a new issue to refactor the handling of optional deps, there are several other places where this decorator should be applied.